### PR TITLE
XRDDEV-994

### PR DESCRIPTION
**Pull request by bertofl90**
Environmental and operational monitoring

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,6 +150,82 @@ For example:
...
+        <xrd:protocolVersion>4.0</xrd:protocolVersion>
+    </SOAP-ENV:Header>
+    <SOAP-ENV:Body>
+        <m:getSecurityServerMetrics/>
+    </SOAP-ENV:Body>
+</SOAP-ENV:Envelope>
+```
+
+#### 1.10.2 Enviornmental monitoring
+
```
Comment (jhyoty) [inline README.md:228]: Looks like this is some kind of copy-paste error? And it is not necessary to copy text from the Environmental monitoring user guide, just link to the documentation.

File: sidecar/Dockerfile [hunk 2]
```diff
@@ -61,7 +64,9 @@ COPY --chown=xroad:xroad files/custom-fetch-pin.sh /usr/share/xroad/autologin/
 RUN sed -i 's/initctl/supervisorctl/g' /usr/share/xroad/scripts/_restore_xroad.sh
 
 RUN mv /usr/share/xroad/jlib/addon/proxy/messagelog.conf /usr/share/xroad/jlib/addon/proxy/messagelog.conf.disabled
+RUN mv /usr/share/xroad/jlib/addon/proxy/opmonitoring.conf /usr/share/xroad/jlib/addon/proxy/opmonitoring.conf.disabled
```
Comment (jhyoty) [inline sidecar/Dockerfile:67]: Combine these to avoid unnecessary layers in the image.

File: README.md [hunk 3]
```diff
@@ -150,6 +150,14 @@ For example:
...
+#### 1.11 Operational monitoring for provider
+
+Operational monitoring for the Security Server Sidecar provider can be  used to obtain information about the services it is running. The operational monitoring processes operational statistics (such as which services have been called, how many times, what was the size of the response, etc.) of the security servers. The operational monitoring will create a database named "op-monitor" for store the data, this database can be configured internally in the container or externally (check 1.6). More information about how to test it can be found here [https://github.com/nordic-institute/X-Road/blob/develop/doc/OperationalMonitoring/Testing/test-opmon_x-road_operational_monitoring_testing_plan_Y-1104-2.md](http://https://github.com/nordic-institute/X-Road/blob/develop/doc/OperationalMonitoring/Testing/test-opmon_x-road_operational_monitoring_testing_plan_Y-1104-2.md "https://github.com/nordic-institute/X-Road/blob/develop/doc/OperationalMonitoring/Testing/test-opmon_x-road_operational_monitoring_testing_plan_Y-1104-2.md")
+
```
Comment (jhyoty) [inline README.md:160]: The links could refer to released docs (master branch). Also, there is a typo in the 1.10. heading.

File: sidecar/Dockerfile [hunk 4]
```diff
@@ -27,11 +27,14 @@ RUN echo "deb $REPO $DIST $COMPONENT" >/etc/apt/sources.list.d/xroad.list \
...
 && apt-get -qq update \
-&& apt-get -qq install xroad-proxy xroad-addon-metaservices xroad-addon-wsdlvalidator xroad-autologin xroad-addon-messagelog \
+&& apt-get -qq install xroad-proxy xroad-addon-metaservices xroad-addon-wsdlvalidator xroad-autologin xroad-addon-messagelog  xroad-addon-opmonitoring  \
 && su - postgres -c "psql postgres -tAc \"DROP DATABASE serverconf;\"" \
 && su - postgres -c "psql postgres -tAc \"DROP USER serverconf;\"" \
 && su - postgres -c "psql postgres -tAc \"DROP DATABASE messagelog;\"" \
 && su - postgres -c "psql postgres -tAc \"DROP USER messagelog;\"" \
+&& su - postgres -c "psql postgres -tAc 'DROP DATABASE \"op-monitor\";'" \
+&& su - postgres -c "psql postgres -tAc \"DROP USER opmonitor;\"" \
+&& pg_ctlcluster 10 main stop \
```
Comment (jhyoty) [inline sidecar/Dockerfile:37]: duplicate of stop on line 36 (causes building the image to fail)

File: sidecar/Dockerfile [hunk 5]
```diff
@@ -27,11 +27,14 @@ RUN echo "deb $REPO $DIST $COMPONENT" >/etc/apt/sources.list.d/xroad.list \
...
 RUN pg_ctlcluster 10 main start \
 && apt-get -qq update \
-&& apt-get -qq install xroad-proxy xroad-addon-metaservices xroad-addon-wsdlvalidator xroad-autologin xroad-addon-messagelog \
+&& apt-get -qq install xroad-proxy xroad-addon-metaservices xroad-addon-wsdlvalidator xroad-autologin xroad-addon-messagelog  xroad-addon-opmonitoring  \
 && su - postgres -c "psql postgres -tAc \"DROP DATABASE serverconf;\"" \
 && su - postgres -c "psql postgres -tAc \"DROP USER serverconf;\"" \
 && su - postgres -c "psql postgres -tAc \"DROP DATABASE messagelog;\"" \
 && su - postgres -c "psql postgres -tAc \"DROP USER messagelog;\"" \
+&& su - postgres -c "psql postgres -tAc 'DROP DATABASE \"op-monitor\";'" \
+&& su - postgres -c "psql postgres -tAc \"DROP USER opmonitor;\"" \
```
Comment (jhyoty) [inline sidecar/Dockerfile:36]: should also drop the "opmonitor_admin" user

File: sidecar/Dockerfile [hunk 6]
```diff
@@ -27,11 +27,14 @@ RUN echo "deb $REPO $DIST $COMPONENT" >/etc/apt/sources.list.d/xroad.list \
 
 RUN pg_ctlcluster 10 main start \
 && apt-get -qq update \
-&& apt-get -qq install xroad-proxy xroad-addon-metaservices xroad-addon-wsdlvalidator xroad-autologin xroad-addon-messagelog \
+&& apt-get -qq install xroad-proxy xroad-addon-metaservices xroad-addon-wsdlvalidator xroad-autologin xroad-addon-messagelog  xroad-addon-opmonitoring  \
```
Comment (raulmartinez-leadin) [inline sidecar/Dockerfile:30]: Consider removing extra blank spaces
```suggestion
&& apt-get -qq install xroad-proxy xroad-addon-metaservices xroad-addon-wsdlvalidator xroad-autologin xroad-addon-messagelog xroad-addon-opmonitoring \
```

File: README.md [hunk 7]
```diff
@@ -150,6 +150,14 @@ For example:
   To install the Security Server Sidecar provider, modify the image build in the setup_security_server_sidecar.sh changing the path "sidecar/Dockerfile" to "sidecar/provider/Dockerfile". The Sidecar provider is based on the Sidecar image and adds support for message loggin, both for internal or remote database (Check 1.6 step).
   To install the Security Server Sidecar provider with finish settings, modify the image build in the setup_security_server_sidecar.sh changing the path "sidecar/Dockerfile" to "sidecar/provider/fi/Dockerfile"
 
+### 1.10 Enviornmental Monitoring for Provider
```
Comment (raulmartinez-leadin) [inline README.md:153]: Typo:
Enviornmental -> Environmental
By the number of hashes (#) looks like a subsection of 1.9, in that case the numbering should be changed to 1.9.1 and so on.
```suggestion
### 1.9.1 Environmental Monitoring for Provider
```

File: README.md [hunk 8]
```diff
@@ -150,6 +150,14 @@ For example:
...
+
+Environmental monitoring for the Security Server Sidecar provider can be  used to obtain information about the platform it's running on, check more information in <https://github.com/nordic-institute/X-Road/blob/master/doc/EnvironmentalMonitoring/Monitoring-architecture.md/>
+
+#### 1.11 Operational monitoring for provider
```
Comment (raulmartinez-leadin) [inline README.md:157]: Consider changing numbering for subsection and adequate subsection title level
```suggestion
### 1.9.2 Operational Monitoring for Provider
```

**Pull request merged**